google-ads-examples/src/main/java/com/google/ads/googleads/examples/misc/UploadImageAsset.java

```diff
@@ -0,0 +1,130 @@
+    } catch (FileNotFoundException fnfe) {
+      System.err.printf(
+          "Failed to load GoogleAdsClient configuration from file. Exception: %s%n", fnfe);
+      return;
+    } catch (IOException ioe) {
+      System.err.printf("Failed to create GoogleAdsClient. Exception: %s%n", ioe);
+      return;
+    }
+
+    try {
+      new UploadImageAsset().runExample(googleAdsClient, params.customerId);
+    } catch (GoogleAdsException gae) {
+      // GoogleAdsException is the base class for most exceptions thrown by an API request.
+      // Instances of this exception have a message and a GoogleAdsFailure that contains a
+      // collection of GoogleAdsErrors that indicate the underlying causes of the
+      // GoogleAdsException.
+      System.err.printf(
+          "Request ID %s failed due to GoogleAdsException. Underlying errors:%n",
+          gae.getRequestId());
+      int i = 0;
+      for (GoogleAdsError googleAdsError : gae.getGoogleAdsFailure().getErrorsList()) {
+        System.err.printf("  Error %d: %s%n", i++, googleAdsError);
+      }
+    }
+  }
+
+  /**
+   * Runs the example.
+   *
+   * @param googleAdsClient the Google Ads API client.
+   * @param customerId the client customer ID.
+   * @throws IOException if there are errors related to image processing.
+   */
+  private void runExample(GoogleAdsClient googleAdsClient, long customerId) throws IOException {
+
+    byte[] imageData = ByteStreams.toByteArray(new URL(IMAGE_URL).openStream());
+
+    // Create the image asset.
+    ImageAsset imageAsset = ImageAsset.newBuilder()
+        .setData(BytesValue.of(ByteString.copyFrom(imageData)))
+        .build();
+
+    // Creates an asset.
+    Asset asset = Asset.newBuilder()
+        // Optional: Provide a unique friendly name to identify your asset.
+        // If you specify the name field, then both the asset name and the image being
+        // uploaded should be unique, and should not match another ACTIVE asset in this
+        // customer account.
+        // .setName(StringValue.of("Jupiter Trip # " +  + System.currentTimeMillis()))
+        .setType(AssetType.IMAGE)
+        .setImageAsset(imageAsset)
+        .build();
+
+    // Creates the operation.
+    AssetOperation operation =  AssetOperation.newBuilder().setCreate(asset).build();
+
+    // Creates the service client.
+    try (AssetServiceClient assetServiceClient =
+        googleAdsClient.getLatestVersion().createAssetServiceClient()) {
+      // Issues a mutate request to add the asset.
+      MutateAssetsResponse response =
+          assetServiceClient.mutateAssets(Long.toString(customerId), ImmutableList.of(operation));
+      if (!response.getResultsList().isEmpty()) {
+        System.out.printf("The image asset with resource name '%s' was created.%n",
+            response.getResults(0).getResourceName());
+      } else {
+        System.out.print("No image asset was created.%n");
```

Comment:
Why would this happen? Surely if the asset wasn't created the RPC would fail...

Reply:
I had followed the PHP example but agree with you. Perhaps the behavior is different in PHP. I'll change that.

Reply:
Can you flag this with the PHP owners? I don't think this is library specific - I'm guessing this is unreachable code.

Reply:
@PierrickVoulet ^^

Reply:
I think it is nit. We will keep it as-is for now in PHP to stay consistent with other code examples.